core/src/err.rs

```diff
@@ -1,35 +1,62 @@
-use constants::{GENERAL_ERROR, TIMEOUT_EXPIRED, UNABLE_TO_CONNECT};
+use crate::Error::InvalidCursorState;
+use constants::{
+    GENERAL_ERROR, INVALID_CURSOR_STATE, INVALID_DESCRIPTOR_INDEX, NO_DSN_OR_DRIVER,
+    TIMEOUT_EXPIRED, UNABLE_TO_CONNECT,
+};
 use mongodb::error::{BulkWriteFailure, ErrorKind, WriteFailure};
 use thiserror::Error;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Error, Debug)]
 pub enum Error {
+    #[error(transparent)]
```

Comment:
I made a lot of `transparent` errors as part of this PR. We may want to make a follow-up ticket that examines and improves our error messages.

Reply:
I would suggest filing that ticket now so we don't forget. Seems like a good idea to me.

Reply:
good idea -- just filed